modules/javafx.graphics/src/test/java/test/javafx/animation/AnimationTest.java

```diff
@@ -254,6 +254,23 @@ public void testJumpTo_UNKNOWN() {
         animation.jumpTo(Duration.UNKNOWN);
     }
 
+    @Test
+    public void testJumpTo_IndefiniteCycles() {
+        animation.shim_setCycleDuration(TWO_SECS);
+        animation.setCycleCount(Animation.INDEFINITE);
+
+        animation.jumpTo("end");
+        assertEquals(TWO_SECS, animation.getCurrentTime());
+    }
+
+    @Test
+    public void testJumpTo_IndefiniteCycleDuration() {
+        animation.shim_setCycleDuration(Duration.INDEFINITE);
+
+        animation.jumpTo("end");
+        assertEquals(Duration.millis(Long.MAX_VALUE / 6), animation.getCurrentTime());
```

Comment:
Why `/ 6` ?

Reply:
This is a good question and I think I should have explained it in a comment because it also points to a mini-flaw. The short answer is that the 6 comes from the `TicksCalculation` class, which defines `TICKS_PER_SECOND = 6000` and `TICKS_PER_MILI = TICKS_PER_SECOND / 1000.0` which is 6.

The test works as follows:

This is the "ticks from duration" calculation for jumping to the end (`Duration.INDEFINITE`), demonstrated by mathematical substitutions:
```
double millis = Duration.INDEFINITE.toMillis() = Double.POSITIVE_INFINITY
long ticks = TickCalculation.fromMillis(millis) = Math.round(TICKS_PER_MILI * millis)
        = Math.round(6 * Double.POSITIVE_INFINITY) = Math.round(Double.POSITIVE_INFINITY) 
        = Long.MAX_VALUE (as per the spec. of Math.round)
```
Notice that we lose precision when multiplying `POSITIVE_INFINITY`.

Then getting the current time calculates "duration from ticks":

```
animation.getCurrentTime() = TickCalculation.toDuration(currentTicks)
        = ticks / TICKS_PER_MILI = Long.MAX_VALUE / 6
```

So the division carries out normally (there's no `Long.POSITIVE_INFINITY`), but the multiplication does not.

Maybe we shouldn't convert between the `double`-based `Duration` and the `long` ticks so much, but I think that this is somewhat negligible.

My next patch is refactoring in preparation of the next, heavier, changes, so I will add this explanation on the test.